python/ipywidgets/ipywidgets/widgets/tests/test_interaction.py

```diff
@@ -187,17 +187,33 @@ def test_dict():
         dict(a=5),
         dict(a=5, b='b', c=dict),
     ]:
-        with pytest.raises(TypeError):
-            c = interactive(f, d=d)
-
+        c = interactive(f, d=d)
+        w = c.children[0]
+        check = dict(
+            cls=widgets.Dropdown,
+            description='d',
+            value=next(iter(d.values())),
+            options=d,
+            _options_labels=tuple(d.keys()),
+            _options_values=tuple(d.values()),
+        )
+        check_widget(w, **check)
 
 def test_ordereddict():
     from collections import OrderedDict
     items = [(3, 300), (1, 100), (2, 200)]
     first = items[0][1]
     values = OrderedDict(items)
-    with pytest.raises(TypeError):
-        c = interactive(f, lis=values)
+    c = interactive(f, lis=values)
+    assert len(c.children) == 2
+    d = dict(
+        cls=widgets.Dropdown,
+        value=first,
+        options=values,
+        _options_labels=("3", "1", "2"),
+        _options_values=(300, 100, 200),
+    )
+    check_widgets(c, lis=d)
```

Comment:
The above check uses `**check` while this uses `lis=d`. I'm not sure if I understand the nuance of why that works, but can maybe look again on Tuesday.

Reply:
Great question. One is using check_widget, the other is using check_widgets :). That got me for a bit too. I'll rename check_widgets to check_widget_children to be more clear.